tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/impl/PostServiceImpl.java

```diff
@@ -668,11 +670,51 @@ public Optional<PostEsrEventDto> markPostAsEsrPositionChanged(
         .postEsrEventDtoToPostEsrEvent(postEsrExportedDto);
     newPostEsrEvent.setPost(post);
 
+    boolean isCurrentEventExpired = handPositionReconciliationExpiry(postEsrExportedDto);
+    if (isCurrentEventExpired && newPostEsrEvent.getStatus() == PostEsrEventStatus.RECONCILED) {
+      newPostEsrEvent.setStatus(PostEsrEventStatus.RECONCILED_EXP);
+    }
+
     PostEsrEvent newPost = postEsrEventRepository.save(newPostEsrEvent);
+
     return Optional.ofNullable(
         postEsrEventDtoMapper.postEsrEventToPostEsrEventDto(newPost));
   }
 
+  /**
+   * Check if there are any expired reconciliation events and set them to RECONCILED_EXP.
+   *
+   * @param postEsrExportedDto the current reconciliation event
+   */
+  protected boolean handPositionReconciliationExpiry(PostEsrEventDto postEsrExportedDto) {
+    Long positionNumber = postEsrExportedDto.getPositionNumber();
+    Long currentPostId = postEsrExportedDto.getPostId();
+    LocalDateTime currentReconciledTime = postEsrExportedDto.getEventDateTime();
+    boolean isCurrentEventExpired = false;
+    Set<PostEsrEvent> postEsrEvents = postEsrEventRepository.findPostEsrEventsByPositionNumber(
+        positionNumber);
+
+    Set<PostEsrEvent> expiredPostEsrEvents = postEsrEvents.stream().filter(
+        postEsrEvent -> postEsrEvent.getStatus() == PostEsrEventStatus.RECONCILED
+            && !postEsrEvent.getPost().getId().equals(currentPostId)
```

Comment:
I'm not sure if this condidion is required as we can set all previous reconciled records to EXP.